app/lib/package/export_api_to_cdn.dart

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:clock/clock.dart';
+import 'package:collection/collection.dart';
+import 'package:crypto/crypto.dart';
+import 'package:gcloud/storage.dart';
+import 'package:logging/logging.dart';
+import 'package:meta/meta.dart';
+import 'package:pool/pool.dart';
+import 'package:retry/retry.dart';
+
+import '../shared/datastore.dart';
+import '../shared/storage.dart';
+import '../shared/utils.dart';
+import '../shared/versions.dart';
+import '../task/global_lock.dart';
+import 'backend.dart';
+import 'models.dart';
+
+final Logger _logger = Logger('export_api_to_cdn');
+
+/// The default concurrency to upload API JSON files to the CDN bucket.
+const _defaultUploadPackageApisToCdnBucketConcurrency = 8;
+
+List<String> _cdnObjectNames(String package) => [
+      '$runtimeVersion/api/packages/$package',
+      'current/api/packages/$package',
```

Comment:
I forgot our strategy wrt. ensuring that only the "current" deployment writes to `current`... how does that work?

Reply:
IIRC every deployment should write to `current/` and we think that it will be fine.